### DISCUSSION TITLE
Master taints

### DISCUSSION THREAD

**Discussion by joaopedrocg27** [techno-tim/k3s-ansible#548] [Questions] [answered]
<img width="530" alt="image" src="https://github.com/user-attachments/assets/81516cda-eb01-4b0e-be76-0c7a23338e7b">

this will set NoSchedule right? Could we make it configurable to change to PreferNoSchedule ?

**Comment by rholmboe** [comment, marked as answer]
Variable `k3s_master_taint` will be used here ([link](https://github.com/techno-tim/k3s-ansible/blob/master/inventory/sample/group_vars/all.yml#L59C1-L69C20)) in all.yml

```yaml
extra_server_args: >-
  {{ extra_args }}
  {{ '--node-taint node-role.kubernetes.io/master=true:NoSchedule' if k3s_master_taint else '' }}
  {% if calico_iface is defined or cilium_iface is defined %}
  --flannel-backend=none
  --disable-network-policy
  --cluster-cidr={{ cluster_cidr | default('10.52.0.0/16') }}
  {% endif %}
  --tls-san {{ apiserver_endpoint }}
  --disable servicelb
  --disable traefik
```

You can change `extra_server_args` variable having `{{ '--node-taint node-role.kubernetes.io/master=true:PreferNoSchedule' if k3s_master_taint else '' }}` instead.
